example/screens/ConnectWithRoomManagerScreen.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useState} from 'react';
+import {
+  Dimensions,
+  Image,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import {useJellyfishClient} from '@jellyfish-dev/react-native-client-sdk';
+import {Button, TextInput, DismissKeyboard} from '../components';
+import {NativeStackScreenProps} from '@react-navigation/native-stack';
+import {AppRootStackParamList} from '../navigators/AppNavigator';
+import type {BottomTabScreenProps} from '@react-navigation/bottom-tabs';
+import {TabParamList} from '../navigators/AppNavigator';
+import {CompositeScreenProps} from '@react-navigation/native';
+import {usePermissionCheck} from '../hooks/usePermissionCheck';
+
+type Props = CompositeScreenProps<
+  BottomTabScreenProps<TabParamList, 'ConnectWithRoomManager'>,
+  NativeStackScreenProps<AppRootStackParamList>
+>;
+
+async function getJellyfishServer(
+  roomManagerUrl: string,
+  roomName: string,
+  userName: string,
+) {
+  const url = roomManagerUrl.endsWith('/')
+    ? roomManagerUrl
+    : roomManagerUrl + '/';
+  const response = await fetch(
+    `${url}api/rooms/${roomName.trim()}/users/${userName.trim()}`,
+  );
+  if (!response.ok) {
+    throw new Error(JSON.stringify(await response.json()));
+  }
+  const tokenData = (await response.json()) as {
+    url: string;
+    token: string;
+  };
+
+  return {
+    jellyfishUrl: tokenData.url,
+    token: tokenData.token,
+  };
+}
+
+const ConnectScreen = ({navigation}: Props) => {
+  const {connect} = useJellyfishClient();
+  const [connectionError, setConnectionError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(false);
+
+  const [roomManagerUrl, setRoomManagerUrl] = useState(
+    process.env.ROOM_MANAGER_URL ?? '',
+  );
+  const [roomName, setRoomName] = useState('');
+  const [userName, setUserName] = useState('');
+
+  usePermissionCheck();
+
+  const onTapConnectButton = async () => {
```

Comment:
useCallback ?

Reply:
It will add only small benefit here - re-rendering button is cheap.